geo/src/algorithm/bool_ops/mod.rs

```diff
@@ -125,6 +126,54 @@ pub enum OpType {
     Xor,
 }
 
+/// Returns the [BooleanOps::union] of all contained geometries
+///
+/// This is more efficient than a manual union, but may result in increased memory use due to the use of an R*-tree
+pub trait UnaryUnion {
+    type Scalar: BoolOpsNum;
+
+    /// Construct a tree of all the input geometries and progressively union them from the "bottom up"
+    ///
+    /// This is considerably more efficient than successively adding new [Polygon]s to an ever more complex [Polygon].
+    /// The output [MultiPolygon] will contain a single member.
```

Comment:
How can you guarantee there will be only one Polygon? What if the two polygons are disjoint?

Reply:
…then there will be more than one Polygon, yes.